x/mobileproxy/README.md

```diff
@@ -0,0 +1,176 @@
+# Mobileproxy: Local Proxy Library for Mobile Apps
+
+This package enables the use Go Mobile to generate a mobile library to run a local proxy and configure your app networking libraries.
+
+### Build the Go Mobile binaries with [`go build`](https://pkg.go.dev/cmd/go#hdr-Compile_packages_and_dependencies)
+
+```bash
+go build -o ./out/ golang.org/x/mobile/cmd/gomobile golang.org/x/mobile/cmd/gobind
+```
+
+### Build the iOS and Android libraries with [`gomobile bind`](https://pkg.go.dev/golang.org/x/mobile/cmd/gomobile#hdr-Build_a_library_for_Android_and_iOS)
+
+```bash
+PATH="$(pwd)/out:$PATH" gomobile bind -target=ios -o "$(pwd)/out/mobileproxy.xcframework" github.com/Jigsaw-Code/outline-sdk/x/mobileproxy
+PATH="$(pwd)/out:$PATH" gomobile bind -target=android -o "$(pwd)/out/mobileproxy.aar" github.com/Jigsaw-Code/outline-sdk/x/mobileproxy
+```
+
+Note: Gomobile expects gobind to be in the PATH, that's why we need to prebuild it, and set up the PATH accordingly.
+
+<details>
+<summary>iOS generated Code</summary>
```

Comment:
The goal is to make it clearer what the actual mobile interfaces look like, since that's what people will be using. It clarifies what the output is too.

It shouldn't diverge if we don't change the API.

Reply:
Committing generated code is generally bad practice, as it diverges and it's hard to tell where the generated file came from. The releases, if we decide to do them, will include the binaries and header files.

The goal of having the code here is educational, so people know what the native library looks like.

Reply:
That's just for illustration, it's not code to be used. People won't be copying and pasting it. They will run the commands in the README instead.

I've updated the text to say "Sample" code to make it clear that may not be the actual code to be used.

Reply:
Or maybe adding "for reference", or "do not use in production" warning before the code snippet would make it more clear?

Reply:
Added a comment and a warning.

Note that there's no risk in accidentally using this in production, since you actually need the XCFramework/AAR with the binaries in it.